### PR TITLE
Only retain the import of subjects

### PR DESCRIPTION
**Pull request by nlisgo**

### CODE REVIEW

File: salt/journal-cms/init.sls [hunk 1]
```diff
@@ -335,43 +335,11 @@ aws-credentials-www-data:
...
-            ../vendor/bin/drush mi jcms_collections_db 2>&1 | tee --append /tmp/drush-migrate.log
-            ../vendor/bin/drush mi jcms_collections_json 2>&1 | tee --append /tmp/drush-migrate.log
-            ../vendor/bin/drush mi jcms_podcast_episodes_json 2>&1 | tee --append /tmp/drush-migrate.log
-            ../vendor/bin/drush mi jcms_covers_db 2>&1 | tee --append /tmp/drush-migrate.log
-            ../vendor/bin/drush mi jcms_covers_json 2>&1 | tee --append /tmp/drush-migrate.log
-            ../vendor/bin/drush mi jcms_highlight_lists_json 2>&1 | tee --append /tmp/drush-migrate.log
             cat /tmp/drush-migrate.log | ../check-drush-migrate-output.sh
-{% elif pillar.elife.env in ['dev'] %}
-    # local VMs do not have the legacy database
-    cmd.run:
```
Comment (giorgiosironi) [inline salt/journal-cms/init.sls]: a small difference is that `dev` seemed not to import subjects before. I'll just run it once to be sure.

Comment (giorgiosironi) [inline salt/journal-cms/init.sls]: it fails locally with 
```
==> journal-cms--vagrant:                   Next Drupal\Core\Database\DatabaseExceptionWrapper: SQLSTATE[42S02]:
==> journal-cms--vagrant:                   Base table or view not found: 1146 Table 'elife_1_0.users' doesn't
==> journal-cms--vagrant:                   exist: SELECT COUNT(*) AS expression
==> journal-cms--vagrant:                   FROM 
==> journal-cms--vagrant:                   (SELECT 1 AS expression
==> journal-cms--vagrant:                   FROM 
==> journal-cms--vagrant:                   {users} u
==> journal-cms--vagrant:                   LEFT OUTER JOIN {users_roles} ur ON ur.uid = u.uid
==> journal-cms--vagrant:                   INNER JOIN {role} r ON r.rid = ur.rid
==> journal-cms--vagrant:                   WHERE u.status = :db_condition_placeholder_0
==> journal-cms--vagrant:                   GROUP BY u.uid) subquery; Array
==> journal-cms--vagrant:                   (
==> journal-cms--vagrant:                       [:db_condition_placeholder_0] => 1
==> journal-cms--vagrant:                   )
==> journal-cms--vagrant:                    in
==> journal-cms--vagrant:                   /ext/srv/journal-cms/web/core/lib/Drupal/Core/Database/Connection.php:685
==> journal-cms--vagrant:                   Stack trace:
==> journal-cms--vagrant:                   #0
==> journal-cms--vagrant:                   /ext/srv/journal-cms/web/core/lib/Drupal/Core/Database/Connection.php(649):
==> journal-cms--vagrant:                   Drupal\Core\Database\Connection->handleQueryException(Object(PDOException),
==> journal-cms--vagrant:                   'SELECT COUNT(*)...', Array, Array)
...
```
so I think this should be left as 2 cases:
- migrate subjects everywhere that is not `dev`
- don't migrate anything on `dev`

Comment (nlisgo) [inline salt/journal-cms/init.sls]: Good catch. When I remove the code that I intend to from `jcms_migrate` I will be able to perform the migration on `dev` also.

**Comment by nlisgo** [comment]
@giorgiosironi I have addressed your feedback

**Pull request merged**